ee/query-service/app/server.go

```diff
@@ -437,7 +438,11 @@ func extractQueryRangeV3Data(path string, r *http.Request) (map[string]interface
 			telemetry.GetInstance().AddActiveLogsUser()
 		}
 		data["dataSources"] = dataSources
-		telemetry.GetInstance().SendEvent(telemetry.TELEMETRY_EVENT_QUERY_RANGE_V3, data, true)
+		ctx := auth.AttachJwtToContext(r.Context(), r)
```

Comment:
I would instead attach this at the beginning of the request.

Reply:
user is not available with this function

Reply:
If it doesn't exist because this middleware is running much earlier, add a middleware that attaches the user/jwt to the context instead of doing it in each endpoint.

Reply:
Yeah it makes sense to add a new middleware, Let me try that.